composer/algorithms/seq_length_warmup/seq_length_warmup.py

```diff
@@ -310,21 +310,16 @@ def _activate_model(self, state: State, logger: Logger) -> None:
                 else:
                     raise
 
-            # Propagate across all ranks if any rank hit CUDA OOM
-            should_handle_cuda_oom = torch.tensor([should_handle_cuda_oom], dtype=torch.uint8, device=device)
-            dist.all_reduce(should_handle_cuda_oom, reduce_operation='MAX')
-            if int(should_handle_cuda_oom.item()) == 1:
-                # If any rank hit CUDA OOM, update grad_accum and retry. Ignore any caught_timeout_error since
-                # it is likely transient, e.g. timeout because certain ranks OOMed and didn't reach barrier.
-                # Raise runtime error if training 1 sample at a time still resulted in CUDA out of memory
-                if state.grad_accum == device_batch_size:
-                    raise RuntimeError(
-                        ('CUDA out of memory. The train loop failed with an internal microbatch of size 1.'
-                         'The GPU does not have enough memory to process even 1 sample.'))
-                else:
-                    state.grad_accum = min(2 * state.grad_accum, device_batch_size)
-                    logger.data_batch({'trainer/grad_accum': state.grad_accum})
-            else:
+            # In-line to avoid circular dependency
+            from composer.trainer.trainer import _get_device, _handle_cuda_oom
+
+            device_arg = None
+            if device is not None:
+                device_type = device.type
+                if 'cuda' in device.type:
+                    device_type = 'gpu'
+                device_arg = _get_device(device_type)
+            if not _handle_cuda_oom(state, should_handle_cuda_oom, device_arg, device_batch_size, is_train=True):
```

Comment:
The side effect is a bit obfuscated here (there's no training happening here, yet `is_train=True`), what about something like:
```
    if device='gpu' and any_rank_oom(cuda_oom):
        adjust_grad_accum(state, device_batch_size)
```